### DISCUSSION TITLE
Next steps

### DISCUSSION THREAD

**Discussion by Buuntu** [Buuntu/react-final-table#44] [Ideas]
This discussion serves to start a conversation around what would be most useful to add to this library for larger adoption.  I don't want to just recreate [React Table](https://react-table.tanstack.com/) with Typescript support, but perhaps there are some other use cases that are sorely needed and would be easy to add?  I also don't want to sacrifice the simplicity and small size of the library.

**Comment by stramel** [comment]
I was thinking of proposing dropping `hidden`.  The main reason I was hesitant about proposing it, is that I'm unsure the use-case of it. I did notice that ReactTable has it

**Reply by Buuntu** [reply]
`hidden` is optional and helps if you just want to feed a dataset to react-table but don't want every single column displayed.  I'd like to keep it for now.

**Reply by stramel** [reply]
>  if you just want to feed a dataset to react-table but don't want every single column displayed

Isn't that just leaving out that column from the column definition array?

> I'd like to keep it for now.

That's fine. I just don't understand the use-case enough

**Reply by Buuntu** [reply]
Yes but what if you want to use that column data for another purpose, such as in the render method of another column, even if you don't want the column itself displayed.  I agree that there are workarounds but this makes it a bit simpler imo.

**Reply by stramel** [reply]
>  if you want to use that column data for another purpose, such as in the render method of another column

Wouldn't the data contain that information already anyways? Or are we stripping it out?

**Reply by cfsdfields** [reply]
I could see phasing out `hidden` columns that are solely meant for including data with allowing columns to derive their data from `row.original` or something similar. So our column's cell value options being either the default 1:1 column-as-key that we have now, or a custom column value that could derive its value from the original data row.

**Comment by stramel** [comment]
It would be nice to be able to provide arbitrary column definitions that aren't directly related to the data structure.

```json
{
  "firstName": "Foo",
  "lastName": "Bar",
}
```

```js
const columns = [{
  name: "fullName",
  label: "Full Name",
  render(_, { firstName, lastName }) {
    return `${firstName} ${lastName}`;
  }
}]
```

I think this could resolve the Action Columns request as well (https://github.com/Buuntu/react-final-table/issues/24)

**Reply by Buuntu** [reply]
Yes, this flexibility would be good to have imo.
